be/src/exprs/vectorized/encryption_functions.cpp

```diff
@@ -199,4 +200,202 @@ ColumnPtr EncryptionFunctions::md5(FunctionContext* ctx, const Columns& columns)
+    result.reserve(size);
+    for (int row = 0; row < size; row++) {
+        if (src_viewer.is_null(row)) {
+            result.append_null();
+            continue;
+        }
+
+        auto src_value = src_viewer.value(row);
+        SHA512Digest digest;
+        digest.update(src_value.data, src_value.size);
+        digest.digest();
+
+        result.append(Slice(digest.hex().c_str(), digest.hex().size()));
+    }
+
+    return result.build(ColumnHelper::is_all_const(columns));
+}
+
+ColumnPtr EncryptionFunctions::sha2(FunctionContext* ctx, const Columns& columns) {
+    if (!ctx->is_constant_column(1)) {
+        auto src_viewer = ColumnViewer<TYPE_VARCHAR>(columns[0]);
+        auto length_viewer = ColumnViewer<TYPE_INT>(columns[1]);
+
+        ColumnBuilder<TYPE_VARCHAR> result;
+
+        auto size = columns[0]->size();
+        result.reserve(size);
+
+        for (int row = 0; row < size; row++) {
+            if (src_viewer.is_null(row)) {
+                result.append_null();
+                continue;
+            }
+
+            auto src_value = src_viewer.value(row);
+            auto length = length_viewer.value(row);
+
+            if (length == 224) {
+                SHA224Digest digest;
+                digest.update(src_value.data, src_value.size);
+                digest.digest();
+                result.append(Slice(digest.hex().c_str(), digest.hex().size()));
+            } else if (length == 0 || length == 256) {
+                SHA256Digest digest;
+                digest.update(src_value.data, src_value.size);
+                digest.digest();
+                result.append(Slice(digest.hex().c_str(), digest.hex().size()));
+            } else if (length == 384) {
+                SHA384Digest digest;
+                digest.update(src_value.data, src_value.size);
+                digest.digest();
+                result.append(Slice(digest.hex().c_str(), digest.hex().size()));
+            } else if (length == 512) {
+                SHA512Digest digest;
+                digest.update(src_value.data, src_value.size);
+                digest.digest();
+                result.append(Slice(digest.hex().c_str(), digest.hex().size()));
+            } else {
+                result.append_null();
+            }
+        }
+
+        return result.build(ColumnHelper::is_all_const(columns));
+    }
+
+    ColumnPtr column = ctx->get_constant_column(1);
+    auto hash_length = ColumnHelper::get_const_value<TYPE_INT>(column);
+
+    if (hash_length != 0 && hash_length != 224 && hash_length != 256 && hash_length != 384 && hash_length != 512) {
+        auto src_viewer = ColumnViewer<TYPE_VARCHAR>(columns[0]);
+        ColumnBuilder<TYPE_VARCHAR> result;
+
+        auto size = columns[0]->size();
+        result.reserve(size);
+
+        for (int row = 0; row < size; row++) {
+            result.append_null();
+        }
+
+        return result.build(ColumnHelper::is_all_const(columns));
```

Comment:
You can call `ColumnHelper::create_const_null_column(size);` to return all null column directly.

Reply:
ok, i will modify it immediately.